src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -332,15 +332,62 @@ internal async Task<Dictionary<string, string>> GetProcessEnvironmentAsync(Cance
             return await helper.ReadEnvironmentAsync(response.Continuation, token).ConfigureAwait(false);
         }
 
-        internal void ApplyStartupHook(string startupHookPath)
+        /// <summary>
+        /// Loads the specified assembly with a StartupHook in the target process.
+        /// </summary>
+        /// <param name="startupHookPath">The path to the assembly containing the StartupHook.</param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="startupHookPath"/> is null or empty.</exception>
+        /// <exception cref="FileNotFoundException">Thrown when the specified assembly does not exist.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the runtime version is less than 8.0.</exception>
+        public void ApplyStartupHook(string startupHookPath)
         {
+            if (string.IsNullOrEmpty(startupHookPath))
+            {
+                throw new ArgumentNullException(nameof(startupHookPath));
+            }
+
+            if (!File.Exists(startupHookPath))
```

Comment:
These APIs cannot validate that the file exists in the local file system. The file may exist in a differently mounted path in a file system from the perspective of the target application and that may not exist in the file system of the process that is calling this method.

Reply:
Oh right, good point, fixed via 29e0161